### DISCUSSION TITLE
How to enable netplay?

### DISCUSSION THREAD

**Discussion by Nailington** [EmulatorJS/EmulatorJS#432] [General]
Hello to the people here!

Me and a friend of mine saw that netplay was added (YAY), and so updated our version of EmulatorJS that we have. Upon updating we are unable to figure out how to turn on netplay; or if it is on, how to use it. I tried looking at the docs, but it hasn't been updated yet. I tried looking at your demo(s) and noticed only the netplay test game actually has it, and not the normal demo on the site, despite them being the same version. I even tried downloading the Mega Mountain game and uploading it to our version of the demo page with no luck, and then tried to add it as a "preloaded game" (where you click the blue Play Game button). Same outcome.

Is there anything we have to do to enable it on our end? Is there limitations on the systems and roms I can use with it? Thank you for any help!

**Comment by ethanaobrien** [comment]
There are 2 additional options you must set. The docs should be updated, you can view the netplay docs here - https://emulatorjs.org/docs4devs/Netplay.html  let me know if you need any help. Thank you!

**Reply by Nailington** [reply]
Thanks! I see this page now! I am curious: Must I use the old cores still? I know the old cores were required a while back, but checking as I would like to use the new cores if possible. 

If I do need to use the old cores, would saves be transferable between the old and new cores, either automatically or via the save exporter?

**Reply by ethanaobrien** [reply]
Yes, currently only the old cores are supported, the new cores are being worked on.

Save states are (mostly) transferable between the old and new cores, however they are not backwards compatible. Im very sorry about this

**Reply by Smevy** [reply]
I am wondering for  all the steps I would like to know what I all need on my computer in order to host this server. I already have NodeJS but I do not know If I need anything else. Also kind of unrelated but on the emulatorjs netplay respatory if you could say what you need installed in order to host a netplay server.

**Reply by ethanaobrien** [reply]
@Smevy the only thing required is nodejs. And just as https://emulatorjs.org/docs4devs/Netplay.html says, run `npm install` and then `npm start`

**Reply by Smevy** [reply]
I tried that but it seemed not to work when I cloned the respatory I typed npm install it it threw errors at me

**Reply by ethanaobrien** [reply]
It would be helpful if you provided those errors...

**Reply by Smevy** [reply]
I have it cloned already but after typing npm install I get these errors
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path C:\Users\Evanj/package.json
npm ERR! errno -4058
npm ERR! enoent ENOENT: no such file or directory, open 'C:\Users\Evanj\package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent


all I know is that npm is not able to find a file if you need anymore I think I might beable to find a file
when I type npm run start same errors are given although I just found a package-lock.json in the directory
C:\Users\USERNAME\

**Reply by ethanaobrien** [reply]
You need to be in the same directory as the project.

**Comment by Smevy** [comment]
Thank you that fixed

**Comment by Smevy** [comment]
Wait actally when putting in the link for the javascript what link do I use or do I use the port
